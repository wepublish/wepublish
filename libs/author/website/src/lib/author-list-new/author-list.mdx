import {Meta, Story, Canvas} from '@storybook/blocks'
import * as stories from './author-list.stories'

<Meta of={stories} />

# Author List

<Canvas>
  <Story of={stories.Default} />
</Canvas>
