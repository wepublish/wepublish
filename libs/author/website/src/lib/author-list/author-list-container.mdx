import {Meta, Story, Canvas, CodeOrSourceMdx} from '@storybook/blocks'
import * as stories from './author-list-container.stories'
import LinkTo from '@storybook/addon-links/react'

<Meta of={stories} />

# Author List Container

> The Author List Container gets authors and renders it as an <LinkTo kind="Components/AuthorList" story="Default">Author component</LinkTo>.

## Override `AuthorListItem`

To replace the `AuthorListItem` component that is being rendered by a custom component, one can override it like this:

<CodeOrSourceMdx>
```typescript
import {WebsiteBuilderProvider, BuilderAuthorListItemProps} from '@wepublish/website'

const MyAuthorListItem = (props: BuilderAuthorListItemProps) => <div>My Author List Item</div>

export default function () {
  return (
    <WebsiteBuilderProvider Author={MyAuthorListItem}>
      <AuthorListContainer />
    </WebsiteBuilderProvider>
  )
}
```
</CodeOrSourceMdx>

<Canvas>
  <Story of={stories.Default} />
</Canvas>
