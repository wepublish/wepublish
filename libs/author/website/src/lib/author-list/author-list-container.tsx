import {AuthorListQuery} from '@wepublish/website/api'
import {QueryResult} from '@apollo/client'
import {useEffect} from 'react'
import {useAuthorListQuery} from '@wepublish/website/api'
import {
  BuilderContainerProps,
  useWebsiteBuilder,
  BuilderAuthorListProps
} from '@wepublish/website/builder'

export type AuthorListContainerProps = {
  onQuery?: (
    queryResult: Pick<QueryResult<AuthorListQuery>, 'data' | 'loading' | 'error' | 'fetchMore'>
  ) => void
} & BuilderContainerProps &
  Pick<BuilderAuthorListProps, 'variables' | 'onVariablesChange'>

export function AuthorListContainer({
  onQuery,
  className,
  variables,
  onVariablesChange
}: AuthorListContainerProps) {
  const {AuthorList} = useWebsiteBuilder()
  const {data, loading, error, fetchMore} = useAuthorListQuery({
    variables
  })

  useEffect(() => {
    onQuery?.({data, loading, error, fetchMore})
  }, [data, loading, error, fetchMore, onQuery])

  return (
    <AuthorList
      data={data}
      loading={loading}
      error={error}
      className={className}
      variables={variables}
      onVariablesChange={onVariablesChange}
    />
  )
}
