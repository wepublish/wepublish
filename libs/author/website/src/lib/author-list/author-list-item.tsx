import {Author} from '@wepublish/website/api'
import {Theme, styled, css, useTheme} from '@mui/material'
import {useWebsiteBuilder} from '@wepublish/website/builder'

export const AuthorListItemImageWrapper = styled('div')`
  display: grid;
  width: 100%;
  aspect-ratio: 1;
`

export const AuthorListItemContent = styled('div')``

const wrapperStyles = (theme: Theme) => css`
  display: grid;
  grid-template-rows: max-content 1fr;
  gap: ${theme.spacing(3)};
  width: 240px;
  text-align: center;
  text-decoration: none;
  color: inherit;
`

const imageStyles = css`
  border-radius: 50%;
`

export function AuthorListItem({
  className,
  slug,
  image,
  name,
  jobTitle
}: Author & {className?: string}) {
  const theme = useTheme()
  const {
    elements: {Image, Paragraph, H6, Link}
  } = useWebsiteBuilder()

  return (
    <Link css={wrapperStyles(theme)} className={className} href={slug}>
      <AuthorListItemImageWrapper>
        {image && <Image image={image} square css={imageStyles} />}
      </AuthorListItemImageWrapper>

      <AuthorListItemContent>
        <H6>{name}</H6>

        {jobTitle && <Paragraph gutterBottom={false}>{jobTitle}</Paragraph>}
      </AuthorListItemContent>
    </Link>
  )
}
