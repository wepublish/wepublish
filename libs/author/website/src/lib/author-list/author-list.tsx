import {styled} from '@mui/material'
import {BuilderAuthorListProps, useWebsiteBuilder} from '@wepublish/website/builder'

export const AuthorListWrapper = styled('article')`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(240px, 1fr));
  gap: ${({theme}) => theme.spacing(4)};
  justify-items: center;
`

export const AuthorList = ({data, className}: BuilderAuthorListProps) => {
  const {AuthorListItem} = useWebsiteBuilder()

  return (
    <AuthorListWrapper className={className}>
      {data?.authors?.nodes.map(author => (
        <AuthorListItem key={author.id} {...author} />
      ))}
    </AuthorListWrapper>
  )
}
