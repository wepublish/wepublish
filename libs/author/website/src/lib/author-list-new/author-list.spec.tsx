import {render} from '@testing-library/react'
import * as stories from './author-list.stories'
import {composeStories} from '@storybook/react'

const storiesCmp = composeStories(stories)

describe('AuthorList', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const {asFragment} = render(<Component />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
