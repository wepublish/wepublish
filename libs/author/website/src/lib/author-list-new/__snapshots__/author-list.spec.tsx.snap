// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorList should render Default 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  >
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
  </article>
</DocumentFragment>
`;

exports[`AuthorList should render WithClassName 1`] = `
<DocumentFragment>
  <article
    class="extra-classname css-13fkobl"
  />
</DocumentFragment>
`;

exports[`AuthorList should render WithEmotion 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  />
</DocumentFragment>
`;

exports[`AuthorList should render WithError 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  />
</DocumentFragment>
`;

exports[`AuthorList should render WithFilters 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  >
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
  </article>
</DocumentFragment>
`;

exports[`AuthorList should render WithLoading 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  />
</DocumentFragment>
`;

exports[`AuthorList should render WithoutImage 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  >
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      />
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      />
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      />
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      />
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      />
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
  </article>
</DocumentFragment>
`;

exports[`AuthorList should render WithoutJobTitle 1`] = `
<DocumentFragment>
  <article
    class="css-13fkobl"
  >
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
      </div>
    </a>
    <a
      class="css-1ge4v1m"
      href="example-slug"
    >
      <div
        class="css-zfebdm"
      >
        <img
          alt="DSC07717"
          class="css-1qnf57e"
          height="6000"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title=""
          width="4000"
        />
      </div>
      <div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Example Name
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Co-Geschäftsleitung & Chefredaktor
        </p>
      </div>
    </a>
  </article>
</DocumentFragment>
`;
